### PR TITLE
x-pack: fix dropped errors

### PR DESCRIPTION
**Pull request by alrs**
This PR fixes some dropped errors in tests under the `x-pack` hierarchy.

### CODE REVIEW

**Comment by fearful-symmetry** [comment]
jenkins, test this

**Comment by exekias** [comment]
jenkins, test this please

ping @narph

File: x-pack/metricbeat/module/azure/compute_vm/compute_vm_test.go [hunk 1]
```diff
@@ -63,6 +63,9 @@ func TestFetch(t *testing.T) {
 		t.Fatal(err)
 	}
 	module, metricsets, err = mb.NewModule(c, mb.Registry)
+	if err != nil {
```
Comment (kaiyan-sheng) [inline x-pack/metricbeat/module/azure/compute_vm/compute_vm_test.go]: How about using `assert.Error()` here?

Comment (narph) [reply x-pack/metricbeat/module/azure/compute_vm/compute_vm_test.go]: assert.Error asserts that a function returned an error but in this case we do not expect one.

**Comment by narph** [comment]
@alrs, do you mind adding an entry in the CHANGELOG as well with a short description of the changes, else we can do it for you.

**Comment by alrs** [comment]
CHANGELOG updated, @narph

File: x-pack/metricbeat/module/azure/compute_vm/compute_vm_test.go (hunk 1 above)
Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/azure/compute_vm/compute_vm_test.go]: Yes that case we can use assert. NoError. But anyway, using `if err!=nil` here works too.

**Pull request merged**